programming_examples/lit.cfg.py

```diff
@@ -143,46 +143,76 @@
 llvm_config.with_environment("PATH", config.aie_tools_dir, append_path=True)
 llvm_config.with_environment("PATH", config.air_tools_dir, append_path=True)
 
-if not config.enable_chess_tests:
-    print("Chess tests disabled")
-else:
-    print("Looking for Chess...")
-
-    result = shutil.which("xchesscc")
-    if result != None:
-        print("Chess found: " + result)
-        config.available_features.add("chess")
-        config.available_features.add("valid_xchess_license")
-        lm_license_file = os.getenv("LM_LICENSE_FILE")
-        if lm_license_file != None:
-            llvm_config.with_environment("LM_LICENSE_FILE", lm_license_file)
-        xilinxd_license_file = os.getenv("XILINXD_LICENSE_FILE")
-        if xilinxd_license_file != None:
-            llvm_config.with_environment("XILINXD_LICENSE_FILE", xilinxd_license_file)
-
-        # test if LM_LICENSE_FILE valid
-        validate_chess = False
-        if validate_chess:
-            import subprocess
-
-            result = subprocess.run(
-                ["xchesscc", "+v"], stdout=subprocess.PIPE, stderr=subprocess.PIPE
-            )
-            validLMLicense = len(result.stderr.decode("utf-8")) == 0
-        else:
-            validLMLicense = lm_license_file or xilinxd_license_file
+config.substitutions.append(("%LLVM_TOOLS_DIR", config.llvm_tools_dir))
 
-        if not lm_license_file and not xilinxd_license_file:
-            print(
-                "WARNING: no valid xchess license that is required by some of the lit tests"
+tool_dirs = [config.aie_tools_dir, config.llvm_tools_dir]
+
+# Test to see if this mlir-air installation is built for peano. If not, then look for Chess in system.
+try:
+    result = subprocess.run(
+        [os.path.join(config.peano_tools_dir, "llc"), "-mtriple=aie", "--version"],
+        stdout=subprocess.PIPE,
+        stderr=subprocess.PIPE,
+    )
+    if re.search("Xilinx AI Engine", result.stdout.decode("utf-8")) is not None:
+        config.available_features.add("peano")
+        config.substitutions.append(("%PEANO_INSTALL_DIR", config.peano_tools_dir))
+        print("Peano found: " + os.path.join(config.peano_tools_dir, "llc"))
+        tool_dirs.append(os.path.join(config.peano_tools_dir, "bin"))
+        # Peano compiler flags
+        peano_flags = (
+            "-O2 -std=c++20 --target=aie2-none-unknown-elf -DNDEBUG -I{}".format(
```

Comment:
Best not to hardcode aie2 here. And most tests already have `--target=aie2-none-unknown-elf %peano_flags`

Reply:
Good point. Hard-coded flag removed.